src/torch_models.py

```diff
@@ -295,305 +502,102 @@ def predict_sample(self, x: Array, rng: np.random.Generator) -> Array:
-        assert x.ndim == 1
-        if self._do_single_class_prediction:
-            classification = self._predicted_single_class
-        else:
-            x = self.normalize(x)
-            classification = self._classify(x)
-        assert classification in [False, True]
-        return classification
 
-    @staticmethod
-    def _normalize_data(data: Array) -> Tuple[Array, Array, Array]:
-        shift = np.min(data, axis=0)  # type: ignore
-        scale = np.max(data - shift, axis=0)  # type: ignore
-        scale = np.clip(scale, CFG.normalization_scale_clip, None)
-        return (data - shift) / scale, shift, scale
-
-    def normalize(self, x: Array) -> Array:
-        """Apply shift and scale to the input."""
-        return (x - self._input_shift) / self._input_scale
-
-    def predict_proba(self, X: Array) -> Array:
-        """Get the predicted probability that the input classifies to 1."""
-        return self(X).detach().numpy()
-
-    def _classify(self, x: Array) -> bool:
-        return self(x).item() > 0.5
-
-    def _fit(self, inputs: Array, outputs: Array) -> None:
-        # Convert data to torch
-        X = torch.from_numpy(np.array(inputs, dtype=np.float32))
-        y = torch.from_numpy(np.array(outputs, dtype=np.float32))
-        # Train
-        logging.info(f"Training {self.__class__.__name__} on {X.shape[0]} "
-                     "datapoints")
-        self.train()  # switch to train mode
-        itr = 0
-        best_loss = float("inf")
-        best_itr = 0
-        n_iter_no_change = CFG.mlp_classifier_n_iter_no_change
-        model_name = tempfile.NamedTemporaryFile(delete=False).name
-        loss_fn = nn.BCELoss()
-        optimizer = optim.Adam(self.parameters(), lr=CFG.learning_rate)
-        while True:
-            yhat = self(X)
-            loss = loss_fn(yhat, y)
-            if loss.item() < best_loss:
-                best_loss = loss.item()
-                best_itr = itr
-                # Save this best model
-                torch.save(self.state_dict(), model_name)
-            if itr % 1000 == 0:
-                logging.info(f"Loss: {loss:.5f}, iter: {itr}/{self._max_itr}")
-            optimizer.zero_grad()
-            loss.backward()
-            optimizer.step()
-            if itr == self._max_itr:
-                break
-            if itr - best_itr > n_iter_no_change:
-                logging.info(f"Loss did not improve after {n_iter_no_change} "
-                             f"itrs, terminating at itr {itr}.")
-                break
-            itr += 1
-        # Load best model
-        self.load_state_dict(torch.load(model_name))  # type: ignore
-        os.remove(model_name)
-        self.eval()  # switch to eval mode
-        yhat = self(X)
-        loss = loss_fn(yhat, y)
-        logging.info(f"Loaded best model with loss: {loss:.5f}")
-
-
-class MLPClassifierEnsemble(Classifier):
+class MLPClassifierEnsemble(BinaryClassifier):
```

Comment:
is it on purpose that this inherits from the base class, and not pytorchclassifier?

Reply:
yeah